### DISCUSSION TITLE
Berry: ESP32 - Serial port does not send byte 0x11

### DISCUSSION THREAD

**Discussion by petrdvorakcz** [arendst/Tasmota#19955] [Q&A]
### PROBLEM DESCRIPTION
_A clear and concise description of what the problem is._
ESP32 - Serial port does not send byte 0x11
I used berry script for sending data over serial line

ser = serial(3, 1, 9600, serial.SERIAL_8N1)
ser.write(bytes('11020B01E1')) 

Monitoring of serial port shows only bytes 020B01E1.

(Sequence '11020B01E1' is for IKEA VINDRIKTNING sensor)

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [ ] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [ ] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [ ] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [ ] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [ ] Device used (e.g., Sonoff Basic): _____
- [ ] Tasmota binary firmware version number used: _____
  - [ ] Pre-compiled
  - [ ] Self-compiled
- [ ] Flashing tools used: _____
- [ ] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:


```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [ ] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:


```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
_Steps to reproduce the behavior:_
Send berry script again

ser = serial(3, 1, 9600, serial.SERIAL_8N1)
ser.write(bytes('11020B01E1')) 

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._

I expected bytes 11020B01E1 on the serial output.

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by barbudor** [comment]
![image](https://github.com/arendst/Tasmota/assets/2996042/9ea4840c-71c1-4a12-af44-fd4eac979709)

**Reply by sfromis** [reply]
Seeing the same, using a terminal program in hex mode. The issue may be that your monitoring method is not fully transparent.

**Comment by Wintespe** [comment]
0x11 may used for software flow control
https://en.wikipedia.org/wiki/Software_flow_control

**Comment by petrdvorakcz** [comment]
Ok, I will retest with other tool for capturing.
For me is strange that PM1006 does not send data if received correct sequence ('11020B01E1').


________________________________
Od: Barbudor ***@***.***>
Odesláno: úterý 7. listopadu 2023 20:52
Komu: arendst/Tasmota ***@***.***>
Kopie: petrdvorakcz ***@***.***>; Author ***@***.***>
Předmět: Re: [arendst/Tasmota] Berry: ESP32 - Serial port does not send byte 0x11 (Discussion #19955)


[image]<https://user-images.githubusercontent.com/2996042/281165090-9ea4840c-71c1-4a12-af44-fd4eac979709.png>

**Comment by barbudor** [comment]
Just out of curiosity, why using Berry on the VINDRIKTNING instead of the native Tasmota driver?

**Comment by petrdvorakcz** [comment]
Native VINDRIKTNING driver is able to send cmd for reading data from pm1006? I found that it can read data only.

Odesláno z aplikace Outlook pro iOS<https://aka.ms/o0ukef>
________________________________
Od: Barbudor ***@***.***>
Odesláno: středa, listopadu 8, 2023 1:04 odp.
Komu: arendst/Tasmota ***@***.***>
Kopie: petrdvorakcz ***@***.***>; Author ***@***.***>
Předmět: Re: [arendst/Tasmota] Berry: ESP32 - Serial port does not send byte 0x11 (Discussion #19955)


Just out of curiosity, why using Berry on the VINDRIKTNING instead of the native Tasmota driver?

**Reply by barbudor** [reply]
It's only reading in spy mode because the read command is sent by the original MCU

So you have disconnected the original MCU?
Does that allow some features that are not available from the native driver?

**Comment by petrdvorakcz** [comment]
Yes, I disconnected pm1006 from original PCB (I bought new PCB https://www.laskakit.cz/laskakit-esp-vindriktning-esp-32-i2c/) and I want tu use berry script to send cmd to pm1006. This feature missing in original driver.
[https://cdn.myshoptet.com/usr/www.laskakit.cz/user/shop/big/7833-6_la100053-esp-vindriktning-top.jpg?63f72cc4]<https://www.laskakit.cz/laskakit-esp-vindriktning-esp-32-i2c/>
LaskaKit ESP-VINDRIKTNING ESP-32 I2C | LaskaKit<https://www.laskakit.cz/laskakit-esp-vindriktning-esp-32-i2c/>
LaskaKit ESP-VINDRIKTNING ESP-32 I2C
www.laskakit.cz

________________________________
Od: Barbudor ***@***.***>
Odesláno: středa 8. listopadu 2023 13:16
Komu: arendst/Tasmota ***@***.***>
Kopie: petrdvorakcz ***@***.***>; Author ***@***.***>
Předmět: Re: [arendst/Tasmota] Berry: ESP32 - Serial port does not send byte 0x11 (Discussion #19955)


It's only reading in spy mode because the read command is sent by the original MCU

So you have disconnected the original MCU?
Does that allow some features that are not available from the native driver?

**Reply by barbudor** [reply]
Nice board but adding 20€ on top of a 10€ device seems a bit overkill to me

**Comment by petrdvorakcz** [comment]
Yes, but with new PCB is it possible to switch off LED in the night. It is possible to address each LED independed to show status of pm1006, CO sensor ... I will try berry script.
Thank you
________________________________
Od: Barbudor ***@***.***>
Odesláno: středa 8. listopadu 2023 13:37
Komu: arendst/Tasmota ***@***.***>
Kopie: petrdvorakcz ***@***.***>; Author ***@***.***>
Předmět: Re: [arendst/Tasmota] Berry: ESP32 - Serial port does not send byte 0x11 (Discussion #19955)


Nice board but adding 20€ on top of a 10€ device seems a bit overkill to me
